src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java

```diff
@@ -0,0 +1,400 @@
+package de.tum.in.www1.artemis.service.iris;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.ws.rs.ForbiddenException;
+
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.event.EventListener;
+import org.springframework.core.env.Profiles;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.iris.IrisTemplate;
+import de.tum.in.www1.artemis.domain.iris.settings.IrisSettings;
+import de.tum.in.www1.artemis.domain.iris.settings.IrisSubSettings;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.repository.iris.IrisSettingsRepository;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+
+/**
+ * Service for managing {@link IrisSettings}.
+ * It is used to manage and combine global, course and exercise specific settings.
+ */
+@Service
+public class IrisSettingsService {
+
+    private final CourseRepository courseRepository;
+
+    private final ApplicationContext applicationContext;
+
+    private final IrisSettingsRepository irisSettingsRepository;
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    public IrisSettingsService(CourseRepository courseRepository, ApplicationContext applicationContext, IrisSettingsRepository irisSettingsRepository,
+            ProgrammingExerciseRepository programmingExerciseRepository) {
+        this.courseRepository = courseRepository;
+        this.applicationContext = applicationContext;
+        this.irisSettingsRepository = irisSettingsRepository;
+        this.programmingExerciseRepository = programmingExerciseRepository;
+    }
+
+    /**
+     * Hooks into the {@link ApplicationReadyEvent} and creates the global IrisSettings object on startup.
+     *
+     * @param event Specifies when this method gets called and provides the event with all application data
+     */
+    @EventListener
+    public void execute(ApplicationReadyEvent event) throws Exception {
+        var settingsOptional = irisSettingsRepository.findAllGlobalSettings();
+        if (settingsOptional.size() == 1) {
+            return;
+        }
+        else if (settingsOptional.size() > 1) {
+            var maxIdSettings = settingsOptional.stream().max(Comparator.comparingLong(IrisSettings::getId)).orElseThrow();
```

Comment:
Note: orElseThrow() will never throw since settingsOptional.size > 1 and max only returns an empty Optional if the stream was empty